dropwizard-jersey/src/main/java/io/dropwizard/jersey/filter/CharsetFilter.java

```diff
@@ -0,0 +1,35 @@
+package io.dropwizard.jersey.filter;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Locale;
+import javax.annotation.Priority;
+import javax.ws.rs.Priorities;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.ext.Provider;
+
+/**
+ * This class ensures that any HTTP response that includes a Content-Type
+ * response header, will also include the UTF-8 character set.
+ */
+@Provider
+@Priority(Priorities.HEADER_DECORATOR)
+public class CharsetFilter implements ContainerResponseFilter {
+
+    private static final String UTF_8 = StandardCharsets.UTF_8.displayName(Locale.ENGLISH);
```

Comment:
I've specifically used this in the past to enforce adding a `;charset=utf-8` into JSON responses, so I'm not sure I'd want to default this to whatever `defaultCharset` returns.